### PR TITLE
refactor the transport

### PR DESCRIPTION
**Pull request by clyang82**
ref: https://github.com/stolostron/backlog/issues/26483
Signed-off-by: clyang82 <chuyang@redhat.com>

### CODE REVIEW

**Comment by clyang82** [comment]
The [2 Vulnerabilities](https://sonarcloud.io/project/issues?id=open-cluster-management_hub-of-hubs&pullRequest=259&resolved=false&types=VULNERABILITY) will be fixed in https://github.com/confluentinc/confluent-kafka-go/pull/880

**Comment by clyang82** [comment]
fix the code coverage by adding `./pkg/...` in sonarcloud target - https://github.com/openshift/release/pull/33561

File: agent/pkg/helper/config.go [hunk 1]
```diff
@@ -4,37 +4,19 @@ import (
...
 
-	kafkaclient "github.com/stolostron/multicluster-global-hub/pkg/kafka"
 	commonobjects "github.com/stolostron/multicluster-global-hub/pkg/objects"
+	"github.com/stolostron/multicluster-global-hub/pkg/transport/consumer"
+	"github.com/stolostron/multicluster-global-hub/pkg/transport/producer"
 )
 
 const (
 	defaultK8sClientsPoolSize = 10
 	maxMessageSizeLimit       = 1024 // to make sure that the message size is below 1 MB.
```
Comment (morvencao) [inline agent/pkg/helper/config.go:17]: We have another universal setting about `maxMessageSizeLimit` that have different value(manager use that one) in https://github.com/stolostron/multicluster-global-hub/pull/259/files#diff-bec1e25388ec2ff9d98cec1ea74e3b910b6ea2af92691cc23e2c775b719ab316
Better to remove this one.

Comment (clyang82) [reply agent/pkg/helper/config.go]: Done

**Review by morvencao — COMMENTED** [review]
overall looks good to me.

For the transport, producer and consumer interfaces, we can still make it more clear.
Correct it I'n wrong.

nit: the Transport interface and Producer/Consumer interfaces releation is not clear. I suppose the Producer/Consumer interfaces should extend Transport interface, which are then implemented by KafkaProducer and KafkaConsumer correspondingly.

image

**Comment by morvencao** [comment]
Let's merge this for now, and I can use the latest transport for the status syncer testing.
/LGTM

**Comment by clyang82** [comment]
@morvencao That is exactly correct. Let me address your comment in another PR.

**Pull request merged**